### PR TITLE
Pullrequest code cleanup exception handling

### PR DESCRIPTION
**Pull request by BenSartor**

### CODE REVIEW

File: app/src/main/java/org/simlar/helper/FileHelper.java [hunk 1]
```diff
@@ -90,6 +91,8 @@ private static void copyFileFromPackage(final Context context, final int resourc
 			outputStream.close();
 			inputStream.close();
 			Lg.i("created ", target);
+		} catch (final FileNotFoundException e) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/FileHelper.java]: Maybe this?
`} catch (final FileNotFoundException | IOException e) {`
` Lg.ex(e, "Failed to create: ", target);`

File: app/src/main/java/org/simlar/https/HttpsPost.java [hunk 2]
```diff
@@ -98,6 +99,8 @@ public static HttpsURLConnection createConnection(final String urlPath, final bo
 			return connection;
 		} catch (final MalformedURLException e) {
 			Lg.ex(e, "MalformedURLException");
+		} catch (final ProtocolException e) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/https/HttpsPost.java]: Maybe
`} catch (final MalformedURLException | ProtocolException | IOException e) {`
`  			Lg.ex(e, "Failed to create connection for", urlPath);`

File: app/src/main/java/org/simlar/helper/PermissionsHelper.java [hunk 3]
```diff
@@ -222,6 +223,8 @@ public static boolean needsExternalStoragePermission(final Context context, fina
 			final FileInputStream stream = new FileInputStream(uri.getPath());
 			stream.close();
 			return false;
+		} catch (final FileNotFoundException e) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/PermissionsHelper.java]: `} catch (final FileNotFoundException | IOException e) {`
` return true;`

**Comment by BenSartor** [comment]
Hi @jritzerfeld,

thanks for your comments and yes it looks a lot nicer this way. But it does not compile as e.g. FileNotFoundException extends IOException.

Kind regards,
 Ben

**Pull request merged**